app/models/experiment_statistics_generator.rb

```diff
@@ -0,0 +1,97 @@
+##
+# All methods are invoked with experiment and user as arguments
+class ExperimentStatisticsGenerator
+
+  # @param experiment [Experiment]
+  # @return [Hash] hash containing simulations statistics
+  def self.simulations_statistics(experiment, _=nil)
+    sims_generated, sims_sent, sims_done = experiment.get_statistics
+
+    if sims_generated > experiment.experiment_size
+      Rails.logger.error("FATAL - too many simulations generated for experiment #{experiment.inspect}")
+      sims_generated = experiment.experiment_size
+    end
+
+    if sims_done > experiment.experiment_size
+      Rails.logger.error("FATAL - too many simulations done and sent for experiment #{experiment.inspect}")
+      sims_done = experiment.experiment_size
+    end
+
+    if sims_done + sims_sent > experiment.experiment_size
+      sims_sent = experiment.experiment_size - sims_done
+    end
+
+    #if sims_generated > @experiment.experiment_size
```

Comment:
Oryginalnie było zakomentowane, przeniosłem w inny kod. Nie czuję się upoważniony to zmieniać, bo nie jestem autorem.